backend/app/api/endpoints/mcp_providers.py

```
@@ -0,0 +1,135 @@
+
+
+def _parse_user_preferences(prefs_data: Optional[str]) -> UserPreferences:
+    """Parse user preferences from JSON string or return default"""
+    if not prefs_data:
+        return UserPreferences()
+    try:
+        if isinstance(prefs_data, str):
+            prefs_dict = json.loads(prefs_data)
+            return UserPreferences.model_validate(prefs_dict)
+        return UserPreferences.model_validate(prefs_data)
+    except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)